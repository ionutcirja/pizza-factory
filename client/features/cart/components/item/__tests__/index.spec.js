import React from 'react';
import { shallow } from 'enzyme';
import { Item } from '..';

describe('Item component', () => {
  let propsToRender;
  
  beforeEach(() => {
    propsToRender = {
      id: '1',
      size: 'small',
      price: '10',
      toppings: ['option 1', 'option 2'],
      quantity: 3,
      actions: {
        updateQuantity: jest.fn(),
        removeFromCart: jest.fn(),
      },
      theme: {
        colours: {
          grey: 'string',
          darkBlue: 'string',
          blue: 'string',
        },
      },
    };
  });
  
  describe('render', () => {
    it('should render the size, price, quantity, a remove button'
      + ' and both a increase and decrease quantity buttons', () => {
      const wrapper = shallow(<Item {...propsToRender} />);
      expect(wrapper).toMatchSnapshot();
    });
  
    it('should render a disabled decrease quantity button'
      + ' if quantity prop value is less or equal than 1', () => {
      propsToRender.quantity = 1;
      const wrapper = shallow(<Item {...propsToRender} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
  
  describe('events handlers', () => {
    it('should call updateQuantity prop method on decrease quantity button click', () => {
      const wrapper = shallow(<Item {...propsToRender} />);
      wrapper.find('DecreaseBtn').props().onClick();
      expect(propsToRender.actions.updateQuantity).toHaveBeenCalledWith({
        id: '1',
        value: -1,
      });
    });
  
    it('should call updateQuantity prop method on decrease quantity button click', () => {
      const wrapper = shallow(<Item {...propsToRender} />);
      wrapper.find('IncreaseBtn').props().onClick();
      expect(propsToRender.actions.updateQuantity).toHaveBeenCalledWith({
        id: '1',
        value: 1,
      });
    });
  
    it('should call removeFromCart prop method on remove button click', () => {
      const wrapper = shallow(<Item {...propsToRender} />);
      wrapper.find('RemoveBtn').props().onClick();
      expect(propsToRender.actions.removeFromCart).toHaveBeenCalledWith({
        id: '1',
      });
    });
  });
});
