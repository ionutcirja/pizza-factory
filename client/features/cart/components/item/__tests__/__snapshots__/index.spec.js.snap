// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item component render should render a disabled decrease quantity button if quantity prop value is less or equal than 1 1`] = `
<CartItem
  borderColour="string"
>
  <RemoveBtn
    colour="string"
    onClick={[Function]}
  />
  <Description
    ingredientsColour="string"
    sizeColour="string"
  >
    <span>
      small
    </span>
    <span>
      (option 1, option 2)
    </span>
  </Description>
  <QuantityWrapper
    colour="string"
  >
    <DecreaseBtn
      colour="string"
      disabled={true}
      disabledColour="string"
      onClick={[Function]}
    />
    <Quantity
      colour="string"
    >
      1
    </Quantity>
    <IncreaseBtn
      colour="string"
      disabledColour="string"
      onClick={[Function]}
    />
  </QuantityWrapper>
  <Price
    colour="string"
  >
    10
  </Price>
</CartItem>
`;

exports[`Item component render should render the size, price, quantity, a remove button and both a increase and decrease quantity buttons 1`] = `
<CartItem
  borderColour="string"
>
  <RemoveBtn
    colour="string"
    onClick={[Function]}
  />
  <Description
    ingredientsColour="string"
    sizeColour="string"
  >
    <span>
      small
    </span>
    <span>
      (option 1, option 2)
    </span>
  </Description>
  <QuantityWrapper
    colour="string"
  >
    <DecreaseBtn
      colour="string"
      disabled={false}
      disabledColour="string"
      onClick={[Function]}
    />
    <Quantity
      colour="string"
    >
      3
    </Quantity>
    <IncreaseBtn
      colour="string"
      disabledColour="string"
      onClick={[Function]}
    />
  </QuantityWrapper>
  <Price
    colour="string"
  >
    10
  </Price>
</CartItem>
`;
